test.ts

```diff
@@ -113,3 +114,31 @@ test("it should send completion after the last emission", t => {
   source.emit(2);
 
 });
+
+
+test("it should with pullable source", t => {
+  t.plan(2);
+  const actual = []
+
+  const timeStart = Date.now();
+  const debounceValue = 1000;
+
+  pipe(
+    fromIter([1, 2, 3]),
+    debounce(debounceValue),
+    s => (start, sink) => {
+      if (start !== 0) return;
+      s(0, (st, d) => {
+        if (st === 2) {
+          const exeTime = Date.now() - timeStart;
+          t.ok(exeTime >= debounceValue);
+          t.equals(actual.length, 0);
```

Comment:
I don't think I understand the test here.

Either `forEach` doesn't consume `pullable streams` or the implementation doesn't let any value pass, but anyhow shouldn't the value be 3 here?
Also I don't really understand why the execution time take the `debounceValue` into account. It seems it is ignored in the implementation of the operator.

Can you help me with this?

Reply:
actually the implementation says that `2` is also debounced IF there is a pending timeout - https://github.com/atomrc/callbag-debounce/pull/3/files#diff-12d250fee1d3743813c57c8a1e58a2a7R20

and also it cancels in that case last timeout (the one that would emit 3) to create a new one that handles it (meaning emitting completion)

I don't think it's particularly useful behaviour though, but I've preserved what was already here.